specifyweb/frontend/js_src/lib/components/WorkBench/WbUtils.tsx

```diff
@@ -132,101 +134,102 @@ export function WbUtilsComponent({
-      >
-        <span className="contents" role="search">
+    <div
+      aria-label={wbText.navigation()}
+      className="flex flex-wrap justify-end gap-x-1 gap-y-2"
+      role="toolbar"
+    >
+      <span className="contents" role="search">
+        <div className="flex">
+          <Input.Generic
+            forwardRef={searchRef}
+            aria-label={commonText.searchQuery()}
+            autoComplete="on"
+            placeholder={commonText.search()}
+            spellCheck
+            title={commonText.searchQuery()}
+            type="search"
+            // @ts-expect-error KeyboardEvent type does not match?
+            onKeyDown={handleSearch}
+          />
+        </div>
+        {!isUploaded && hasPermission('/workbench/dataset', 'update') ? (
           <div className="flex">
-            <Input.Generic
-              forwardRef={searchRef}
-              aria-label={commonText.searchQuery()}
+            <Input.Text
+              forwardRef={replaceRef}
+              aria-label={wbText.replacementValue()}
               autoComplete="on"
-              placeholder={commonText.search()}
-              spellCheck
-              title={commonText.searchQuery()}
-              type="search"
-              onKeyDown={handleSearch}
+              placeholder={wbText.replace()}
+              title={wbText.replacementValue()}
+              // @ts-expect-error KeyboardEvent type does not match?
+              onKeyDown={handleReplace}
+              disabled={isReadOnly}
             />
           </div>
-          {!isUploaded && hasPermission('/workbench/dataset', 'update') ? (
-            <div className="flex">
-              <Input.Text
-                forwardRef={replaceRef}
-                aria-label={wbText.replacementValue()}
-                autoComplete="on"
-                placeholder={wbText.replace()}
-                title={wbText.replacementValue()}
-                onKeyDown={handleReplace}
-              />
-            </div>
-          ) : undefined}
-          <span>
-            <WbAdvancedSearch
-              initialSearchPreferences={utils.searchPreferences}
-              onChange={(newSearchPreferences) => {
-                if (
-                  newSearchPreferences.navigation.direction !==
-                  utils.searchPreferences.navigation.direction
-                ) {
-                  // TODO: add workbench or cells to parent component
-                  // cells.flushIndexedCellData = true;
-                }
-                utils.searchPreferences = newSearchPreferences;
-                // TODO: figure out what searchCells with SettingsChange does
-                // if (utils.searchPreferences.search.liveUpdate)
-                //   utils.searchCells({
-                //     key: 'SettingsChange',
-                //   }).catch(softFail);
-              }}
-            />
-          </span>
+        ) : undefined}
+        <span>
+          <WbAdvancedSearch
```

Comment:
is this `span` needed for layout of `WbAdvancedSearch` to display correctly?
if so, the span should be moved inside the `WbAdvancedSearch` as it's a concern of that component rather than this component

if `span` is needed by this component, rather than `WbAdvancedSearch`, then it should stay here.

i.e think of it like this: if you were to re-use `WbAdvancedSearch` in some other component, would you also need to call it inside of `<span>` or not? if so, then span should be inside of `WbAdvancedSearch`